source/client_channel_handler.c

```diff
@@ -41,6 +42,32 @@ static int s_packet_handler_default(
     return aws_raise_error(AWS_ERROR_MQTT_INVALID_PACKET_TYPE);
 }
 
+static void s_on_time_to_ping(struct aws_channel_task *channel_task, void *arg, enum aws_task_status status);
+static void s_schedule_ping(struct aws_mqtt_client_connection *connection) {
+    aws_channel_task_init(&connection->ping_task, s_on_time_to_ping, connection);
+
+    uint64_t schedule_time = 0;
+    aws_channel_current_clock_time(connection->slot->channel, &schedule_time);
+    if (connection->keep_alive_time) {
```

Comment:
Minor, but request_timeout gets initialized to the default value if its options value comes in as zero.  Seems reasonable to do the same thing with this setting so that the time calculation becomes unconditional.

Reply:
good point.

Reply:
nevermind, this doesn't actually apply, because keep alive has meaning on the connect, and reconnect calls so I don't want to redefine it.